Frontend/ui-customer/src/Flow.purs

```diff
@@ -827,18 +826,21 @@ homeScreenFlow = do
         (GlobalState newState) <- getState
         let state = newState.homeScreen
 
-        case state.props.isSource of
-          Just true -> do
-            (GetPlaceNameResp sourceDetailResp) <- getPlaceNameResp (state.props.sourcePlaceId) (state.props.sourceLat) (state.props.sourceLong) (if state.props.isSource == Just false then dummyLocationListItemState else item)
-            let (PlaceName sourceDetailResponse) = (fromMaybe HomeScreenData.dummyLocationName (sourceDetailResp !! 0))
-                (LatLong sourceLocation) = sourceDetailResponse.location
-            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{ props{sourceLat = sourceLocation.lat, sourceLong = sourceLocation.lon} })
-          Just false  -> do
-            (GetPlaceNameResp destinationDetailResp) <- getPlaceNameResp (state.props.destinationPlaceId) (state.props.destinationLat) (state.props.destinationLong) (if state.props.isSource == Just true then dummyLocationListItemState else item)
-            let (PlaceName destinationDetailResponse) = (fromMaybe HomeScreenData.dummyLocationName (destinationDetailResp!!0))
-                (LatLong destinationLocation) = (destinationDetailResponse.location)
-            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{ props{destinationLat = destinationLocation.lat, destinationLong = destinationLocation.lon} })
-          _          -> pure unit
+        case state.props.sourceSelectedOnMap of
```

Comment:
yes @aryan-13 since you implemented caching feature, I have to handle the case for `isSourceSelectedOnMap` in this case otherwise it will show `Current Location` only

Reply:
if there is any other case please mention here, so we can make sure no issue is there related to caching and placeName api calls

Reply:
Sure